runtime/include/chpl-util.h

```diff
@@ -53,4 +60,44 @@ unsigned char* round_up_to_mask_ptr(unsigned char* p, uintptr_t mask)
 }
 
 
-#endif // _chpl_align_h_
+////////////////////////////////////////
+//
+// snprintf() a K/M/G size
+//
+
+static inline
+int chpl_snprintf_z_KMG(char* buf, int bufSize, size_t val) {
+  const size_t GiB = (size_t) (1UL << 30);
+  if (val >= GiB)
+    return snprintf(buf, bufSize, "%zdG", val / GiB);
+
+  const size_t MiB = (size_t) (1UL << 20);
+  if (val >= MiB)
+    return snprintf(buf, bufSize, "%zdM", val / MiB);
+
+  const size_t KiB = (size_t) (1UL << 10);
+  if (val >= KiB)
+    return snprintf(buf, bufSize, "%zdK", val / KiB);
+
+  return snprintf(buf, bufSize, "%zd", val);
+}
+
+static inline
+int chpl_snprintf_f_KMG(char* buf, int bufSize, double val) {
+  const double GiB = (double) (1UL << 30);
+  if (val >= GiB)
+    return snprintf(buf, bufSize, "%.1fG", val / GiB);
+
+  const double MiB = (double) (1UL << 20);
```

Comment:
No longer c90 style decls -- do we still care about avoiding mixed decls and code?

Reply:
I agree, but I think mixed decls would make a lot of runtime code easier to read, but we've historically stayed away from it. I just can't remember if we stopped caring about C90.

Reply:
Ah, it looks like we officially stopped caring in https://github.com/chapel-lang/chapel/pull/6224, so +1 for mixed-decls